src/js/media-controller.js

```diff
@@ -947,7 +955,26 @@ const Delegates = {
 
   [MediaUIAttributes.MEDIA_PLAYBACK_RATE](el) {
     return getPlaybackRate(el);
-  }
+  },
+
+  [MediaUIAttributes.MEDIA_STREAM_TYPE](el) {
+    const media = el.media;
+    const duration = media.duration;
+
+    if (duration === Infinity) {
+      return StreamTypes.LIVE;
+    } else if (!isNaN(duration)) {
+      return StreamTypes.ON_DEMAND;
+    } else {
+      const defaultType = el.getAttribute('default-stream-type');
+
+      if (Object.values(StreamTypes).includes(defaultType)) {
```

Comment:
nitpick(non-blocking): consider pulling out `Object.values(StreamTypes)` since it shouldn't need to be re-evaluated every method invocation.

Reply:
This is a pattern we tend to use on elements (see: https://github.com/muxinc/elements/search?q=Object.values)